### PR TITLE
Prefer scheduling instances on non-network nodes

### PR DESCRIPTION
**Pull request by mandoonandy**
Is this what you were thinking in issue #156?

### CODE REVIEW

File: shakenfist/scheduler.py [hunk 1]
```diff
@@ -209,6 +209,15 @@ def place_instance(self, instance, network, candidates=None):
...
                          'schedule', 'Have most matching images', None, str(candidates))
 
+            # Avoid allocating to network node if possible
+            net_node = db.get_network_node()
+            if len(candidates) > 1 and net_node['fqdn'] in candidates:
+                candidates.remove(net_node['fqdn'])
+                LOG.info('Scheduling %s, %s are non-network nodes'
+                            % (instance, candidates))
+                db.add_event('instance', instance.db_entry['uuid'],
+                             'schedule', ' are non-network nodes', None, str(candidates))
```
Comment (mikalstill) [inline shakenfist/scheduler.py:219]: Sorry to be a pain, but I think that should be "Are non-network nodes" to be consistent with the others here?

File: shakenfist/db.py [hunk 2]
```diff
@@ -50,6 +50,13 @@ def get_nodes():
     return etcd.get_all('node', None)
 
 
+def get_network_node():
+    see_this_node()
+    for n in get_nodes():
+        if n['ip'] == config.parsed.get('NETWORK_NODE_IP'):
+            return n
+
+
```
Comment (mikalstill) [inline shakenfist/db.py:59]: This is fine for here, but it leaves me feeling like the current way of specifying if a node is a network node or not is clunky. I should think more about that.

File: shakenfist/scheduler.py (hunk 1 above)
Comment (mandoonandy) [reply shakenfist/scheduler.py:219]: Absolutely.

**Comment by mikalstill** [comment]
Oh sorry, this was on the wrong branch. We should propose these against master and then cherry pick them to v0.2-devel. Fixing that now.

**Pull request merged**